src/main/java/com/mate/academy/demo/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.mate.academy.demo.service;
+
+import com.mate.academy.demo.dto.OrderDto;
+import com.mate.academy.demo.dto.OrderItemDto;
+import com.mate.academy.demo.dto.OrderStatusRequest;
+import com.mate.academy.demo.exception.DataProcessingException;
+import com.mate.academy.demo.mapper.OrderMapper;
+import com.mate.academy.demo.model.Order;
+import com.mate.academy.demo.model.ShoppingCart;
+import com.mate.academy.demo.repository.OrderRepository;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService cartService;
+    private final OrderMapper mapper;
+    private final OrderRepository repository;
+
+    @Override
+    public OrderDto placeOrder(Long userId, String shippingAddress) {
+        ShoppingCart shoppingCart = cartService.fetchShoppingCartModel(userId);
+        if (shoppingCart.getCartItems() == null
+                || shoppingCart.getCartItems().isEmpty()) {
+            throw new DataProcessingException("Your cart is empty");
```

Comment:
Which response status does this exception return? Better create a separate OrderProcessingException